qiskit_experiments/data_processing/data_processor.py

```diff
@@ -274,6 +274,22 @@ def _data_extraction(self, data: Union[Dict, List[Dict]]) -> np.ndarray:
 
     def __repr__(self):
         """String representation of data processors."""
-        names = ", ".join(node.__class__.__name__ for node in self._nodes)
+        names = ", ".join(repr(node) for node in self._nodes)
 
         return f"{self.__class__.__name__}(input_key={self._input_key}, nodes=[{names}])"
+
+    def __json_encode__(self) -> Dict[str, Any]:
+        """Return the config dict for this data processor."""
+        return dict(
+            cls=type(self),
+            input_key=self._input_key,
+            nodes=self._nodes,
```

Comment:
Does this work? Perhaps I am missing something on how the serialization works but I would expect that we should add a config dict or something similar and not the class instances?

Reply:
It works as you can see in new unittest. Indeed JSON serializer recursively serialize the object. If you pass a dict to serializer, the serializer is recursively called for values.